trunk/SUAVE/Methods/Noise/Fidelity_One/Propeller/compute_broadband_noise.py

```diff
@@ -1,123 +1,349 @@
+        c         = np.tile(blade_chords[None,None,None,:,None,None],(num_cpt,num_mic,num_rot,1,num_azi,num_cf))
+        delta_r   = np.tile(delta_r[None,None,None,:,None,None],(num_cpt,num_mic,num_rot,1,num_azi,num_cf))
+        M         = np.tile(M[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,1))  
+        c_0       = np.tile(c_0[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+        beta_sq   = np.tile(beta_sq[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+        Omega     = np.tile(Omega[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+        U_inf     = np.tile(U_inf[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+        rho       = np.tile(rho[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+        kine_visc = np.tile(kine_visc[:,None,None,None,None,None,:],(1,num_mic,num_rot,num_sec,num_azi,num_cf,2))
+
+        X   = np.repeat(bspv.blade_section_coordinate_sys[:,:,:,:,:,:,0,:],2,axis = 6)
+        Y   = np.repeat(bspv.blade_section_coordinate_sys[:,:,:,:,:,:,1,:],2,axis = 6)
+        Z   = np.repeat(bspv.blade_section_coordinate_sys[:,:,:,:,:,:,2,:],2,axis = 6)
+
+        # ------------------------------------------------------------
+        # ****** BLADE MOTION CALCULATIONS ******
+        # the rotational Mach number of the blade section
+        omega   = np.tile((2*np.pi*frequency)[None,None,None,None,None,:,None],(num_cpt,num_mic,num_rot,num_sec,num_azi,1,2))
+        r       = np.repeat(r[:,:,:,:,:,:,np.newaxis],2,axis = 6)
+        c       = np.repeat(c[:,:,:,:,:,:,np.newaxis],2,axis = 6)/2
+        delta_r = np.repeat(delta_r[:,:,:,:,:,:,np.newaxis],2,axis = 6)
+        M       = np.repeat(M,2,axis = 6)
+        R_s     = np.repeat(np.linalg.norm(bspv.blade_section_coordinate_sys,axis = 6),2,axis = 6)
+        mu      = (omega/(1 +(Omega*r/c_0)*(X/R_s)))*M/(U_inf*beta_sq)
+
+        # ------------------------------------------------------------
+        # ****** LOADING TERM CALCULATIONS ******
+        # equation 7
+        epsilon       = X**2 + (beta_sq)*(Y**2 + Z**2)
+        gamma         = np.sqrt(((mu/epsilon)**2)*(X**2 + beta_sq*(Z**2)))
+        ss_1, cc_1    = fresnel(2*((((omega/(1 +  (Omega*r/c_0)*(X/R_s))) /(0.8*U_inf)) /c) + (mu/c)*M + (gamma/c)))
+        ss_2, cc_2    = fresnel(2*((mu/c)*X/epsilon + (gamma/c)) )
+        triangle      = (omega/(U_inf*c)) - (mu/c)*X/epsilon + (mu/c)*M
+        norm_L_sq     = (1/triangle)*abs(np.exp(1j*2*triangle)*((1 - (1 + 1j)*(cc_1 - 1j*ss_1)) \
+                        + ((np.exp(-1j*2*triangle))*(np.sqrt((((omega/(1 +  (Omega*r/c_0)*(X/R_s))) /(0.8*U_inf)) + mu*M + gamma)/(mu*X/epsilon +gamma))) \
+                           *(1 + 1j)*(cc_2 - 1j*ss_2)) ))
+        norm_L_sq     = np.nan_to_num(norm_L_sq)
```

Comment:
under what conditions does this NaN?

Reply:
the numbers are very very large, like  ~10^40. that is why I cannot lower the precision. Also why on the regression I had to loosen the tolerance (similar to the harmonic noise regression).

Reply:
What I was getting at here was numbers cause a nan? The best way to handle this is to do it in the equations. 

Is it an exp overflow? Is it inf that becomes a nan?

 It's not for coding style it's for optimization. The gradients out of nan_to_num will be useless.

Reply:
Well, I do put a condition later that changes infs to zero so it seems I don't need the nan_2_num